python/az/aro/azext_aro/_aad.py

```diff
@@ -51,6 +54,19 @@ def create_service_principal(self, app_id):
                 logger.warning("%s; retry %d of %d", ex, retries, max_retries)
                 time.sleep(10)
 
+    def delete_service_principal(self, app_id):
+        max_retries = 3
+        retries = 0
+        while True:
+            try:
+                return self.client.service_principal_delete(app_id)
+            except GraphError as ex:
+                if retries >= max_retries:
+                    raise
```

Comment:
Stylistic preference but as per PEP8 "explicit is better than implicit" I'd prefer to see `raise ex` not just `raise`.